src/app/store/chats-store/chats.effects.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { catchError, map, of, switchMap } from 'rxjs';
+import { ChatService } from '../../services/chat.service';
+import {
+  getOldMessages,
+  getOldMessagesComplete,
+  getOldMessagesFailure,
+  sendNewMessage,
+} from './chats.actions';
+
+@Injectable()
+export class ChatEffects {
+  getOldMessages$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(getOldMessages),
+      switchMap(() =>
+        this.chatService.getOldMessages().pipe(
+          map((messages) => {
+            return getOldMessagesComplete({ messages });
+          }),
+          catchError((error) => of(getOldMessagesFailure({ error })))
```

Comment:
Rather handle the error here, alert the user, and then return EMPTY or something